File: nim_core_v2/ios/Classes/FLTConvert.swift
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

import Foundation
import NIMSDK

enum FLTConvertError: Error {
  case runtimeError(String)
}

extension NIMMessageType {
  static func getType(_ sType: String?) throws -> NIMMessageType? {
    switch sType {
    //        case "undef"
    //            return NIMMessageType.undef
    case "text":
      return NIMMessageType.text
    case "image":
      return NIMMessageType.image
    case "audio":
      return NIMMessageType.audio
    case "video":
      return NIMMessageType.video
    case "location":
      return NIMMessageType.location
    case "notification":
      return NIMMessageType.notification
    case "file":
      return NIMMessageType.file
    case "tip":
      return NIMMessageType.tip
    case "robot":
      return NIMMessageType.robot
    case "rtc":
      return NIMMessageType.rtcCallRecord
    case "custom":
      return NIMMessageType.custom
    default:
      break
    }
    throw FLTConvertError
      .runtimeError("MsgTypeEnum not contains \(sType ?? "nuknow NIMSessionType")")
  }
}

extension NIMSessionType {
  static func getType(_ sType: String?) throws -> NIMSessionType? {
    switch sType {
    //        case "none":
    //            return NIMSessionType.P2P
    case "p2p":
      return NIMSessionType.P2P
    case "team":
      return NIMSessionType.team
    case "superTeam":
      return NIMSessionType.superTeam
    case "system":
      return NIMSessionType.YSF
    case "chatRoom":
      return NIMSessionType.chatroom
    default:
      break
    }

    throw FLTConvertError
      .runtimeError("SessionTypeEnum not contains \(sType ?? "nuknow NIMSessionType")")
  }
}

enum NIMNosScene: String {
  /// 用户、群组资料（eg:头像
  case defaultProfile

  /// 私聊、群聊、聊天室发送图片、音频、视频、文件..
  case defaultIm

  /// sdk内部上传文件（eg:日志）并且对应的过期时间不允许修改
  case systemNosScene

  /// 安全文件下载sceneKey前缀， 包含此前缀的都认为是安全文件
  case securityPrefix

  func getScene() -> String {
    switch self {
    case .defaultProfile:
      return NIMNOSSceneTypeAvatar

    case .defaultIm:
      return NIMNOSSceneTypeMessage

    case .systemNosScene:
      return "nim_system"

    case .securityPrefix:
      return NIMNOSSceneTypeSecurity
    }
  }

  static func toDic(scene: String?) -> String {
    switch scene {
    case NIMNOSSceneTypeAvatar:
      return NIMNosScene.defaultProfile.rawValue

    case NIMNOSSceneTypeMessage:
      return NIMNosScene.defaultIm.rawValue
    case "nim_system":
      return NIMNosScene.systemNosScene.rawValue
    case NIMNOSSceneTypeSecurity:
      return NIMNosScene.securityPrefix.rawValue
    default:
      return NIMNosScene.defaultIm.rawValue
    }
  }
}

File: nim_core_v2_windows/windows/src/FLTConvert.cpp
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

#include "FLTConvert.h"

Convert::Convert() {}

std::string Convert::getStringFormMapForLog(
    const flutter::EncodableMap* arguments) const {
  if (!arguments) {
    return "";
  }

  // todo
  return "";
}

void Convert::getLogList(const std::string& strLog,
                         std::list<std::string>& listLog) const {
  listLog.clear();
  int num = 15 * 1024;        // 分割定长大小
  int len = strLog.length();  // 字符串长度
  int end = num;
  for (int start = 0; start < len;) {
    if (end > len)  // 针对最后一个分割串
    {
      listLog.emplace_back(
          strLog.substr(start, len - start));  // 最后一个字符串的原始部分
      break;
    }
    listLog.emplace_back(
        strLog.substr(start, num));  // 从0开始，分割num位字符串
    start = end;
    end = end + num;
  }
}

std::string Convert::getStringFormListForLog(
    const flutter::EncodableList* arguments) const {
  if (!arguments) {
    return "";
  }

  return "";
}
